gulpfile.js/build.js

```diff
@@ -0,0 +1,411 @@
+const {project, travis} = require('@lib/utils');
+const git = require('@lib/utils/git');
+const ComponentReferenceImporter = require('@lib/pipeline/componentReferenceImporter');
+const SpecImporter = require('@lib/pipeline/specImporter');
+// TODO: Fails on Travis with HttpError: Requires authentication
+// const roadmapImporter = require('@lib/pipeline/roadmapImporter');
+const {pageTransformer} = require('@lib/build/pageTransformer');
+const gulpSass = require('gulp-sass');
+const test = require('./test.js');
+
+// The Google Cloud Storage bucket used to store build job artifacts
+const TRAVIS_GCS_PATH = 'gs://amp-dev-ci/travis/';
+// Branches that should actually trigger a build
+const BUILDING_BRANCHES = ['future'];
+
+
+/**
+ * Cleans all directories/files that get created by any of the following
+ * tasks
+ *
+ * @return {Promise}
+ */
+function clean() {
+  return del([
+    project.absolute('.cache/**/*'),
+
+    project.paths.DIST,
+    project.paths.BUILD,
+
+    project.absolute('boilerplate/dist'),
+
+    project.paths.CSS,
+    project.absolute('pages/extensions/**/*.pyc'),
+    project.absolute('pages/content/amp-dev/documentation/examples/documentation/**/*.html'),
+    project.absolute('pages/content/amp-dev/documentation/examples/previews/**/*.html'),
+    project.absolute('pages/icons'),
+    project.absolute('pages/layouts'),
+    project.absolute('pages/macros'),
+    project.absolute('pages/views'),
+    project.absolute('pages/.depcache.json'),
+    project.absolute('pages/podspec.yaml'),
+
+    project.paths.GROW_BUILD_DEST,
+    project.paths.STATICS_DEST,
+    project.absolute('platform/static'),
+
+    project.absolute('playground/dist'),
+  ], {'force': true});
+}
+
+
+/**
+ * Compiles all SCSS partials to CSS
+ *
+ * @return {Stream}
+ */
+function sass() {
+  const options = {
+    'outputStyle': 'compressed',
+    'includePaths': project.paths.SCSS,
+  };
+
+  return gulp.src(`${project.paths.SCSS}/**/[^_]*.scss`)
+      .pipe(gulpSass(options))
+      .on('error', function(e) {
+        console.error(e);
+        // eslint-disable-next-line no-invalid-this
+        this.emit('end');
+      })
+      .pipe(gulp.dest(project.paths.CSS));
+}
+
+/**
+ * Copies the templates into the Grow pod
+ *
+ * @return {Stream}
+ */
+function templates() {
+  return gulp.src(`${project.paths.TEMPLATES}/**/*`)
+      .pipe(gulp.dest(project.paths.GROW_POD));
+}
+
+/**
+ * Copies the icons into the Grow pod
+ *
+ * @return {Stream}
+ */
+function icons() {
+  return gulp.src(`${project.paths.ICONS}/**/*`)
+      .pipe(gulp.dest(`${project.paths.GROW_POD}/icons`));
+}
+
+/**
+ * Runs all tasks needed to build the frontend
+ * @return {undefined}
+ */
+function buildFrontend(done) {
+  return (gulp.parallel(sass, templates, icons))(done);
+}
+
+/**
+ * Builds the playground
+ * @return {Promise}
+ */
+async function buildPlayground() {
+  await sh('mkdir -p playground/dist');
```

Comment:
This is a dumb fix, but strangely the `clean-webpack-plugin` suddenly [failed](https://travis-ci.org/ampproject/docs/jobs/532013888) when the `playground/dist` wasn't there.

```
/home/travis/build/ampproject/docs/node_modules/clean-webpack-plugin/dist/clean-webpack-plugin.js:205
      throw error;
      ^
Error: ENOENT: no such file or directory, stat '/home/travis/build/ampproject/docs/playground/dist'
    at Object.statSync (fs.js:850:3)
```

Couldn't reproduce it locally though. You've got an idea or maybe a package in mind that we updated recently and might cause this?

Reply:
No idea